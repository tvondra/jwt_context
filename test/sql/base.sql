\set ECHO none

-- disable the notices for the create script (shell types etc.)
SET client_min_messages = 'WARNING';
\i sql/jwt_context--1.0.0.sql
SET client_min_messages = 'NOTICE';

\set ECHO all

-- generate new keypair
SELECT * FROM jwt_generate_keys('ES256') \gset

-- generate new secret
SELECT * FROM jwt_generate_secret('HS256') \gset

-- incorrect combinations
SELECT * FROM jwt_generate_keys('HS256') \gset
SELECT * FROM jwt_generate_secret('ES256') \gset

SELECT jwt_sign(NULL, NULL, NULL, NULL);
SELECT jwt_verify(NULL, NULL);

SELECT jwt();
SELECT jwt('id');
SELECT jwt(NULL);
