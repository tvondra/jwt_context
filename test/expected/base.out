\set ECHO none
-- generate new keypair
SELECT * FROM jwt_generate_keys('ES256') \gset
-- generate new secret
SELECT * FROM jwt_generate_secret('HS256') \gset
-- incorrect combinations
SELECT * FROM jwt_generate_keys('HS256') \gset
ERROR:  algorithm 'HS256' does not support 'genkey'
SELECT * FROM jwt_generate_secret('ES256') \gset
ERROR:  algorithm 'ES256' does not support 'gensecret'
SELECT jwt_sign(NULL, NULL, NULL, NULL);
 jwt_sign 
----------
 
(1 row)

SELECT jwt_verify(NULL, NULL);
 jwt_verify 
------------
 
(1 row)

SELECT jwt();
 jwt 
-----
 
(1 row)

SELECT jwt('id');
 jwt 
-----
 
(1 row)

SELECT jwt(NULL);
 jwt 
-----
 
(1 row)

